### PR TITLE
feat: adding unit messaging to controller

### PR DESCRIPTION
**Pull request by Tbaile**
Adding API allowing unit to push for it's changes.
Ref:
- https://github.com/NethServer/nethsecurity/issues/800#issuecomment-2404234577

### CODE REVIEW

File: api/main.go [hunk 1]
```diff
@@ -140,6 +140,7 @@ func main() {
 
 	// Ingest APIs: receive data from firewalls
 	authorized := router.Group("/ingest", middleware.BasicAuth())
+	authorized.POST("/info", methods.ForcePushInfo)
```
Comment (gsanchietti) [inline api/main.go:143]: Why naming Force?
```suggestion
	authorized.POST("/info", methods.PushInfo)
```

File: api/methods/unit.go [hunk 2]
```diff
@@ -143,6 +143,36 @@ func GetUnitInfo(c *gin.Context) {
 
 }
 
+func ForcePushInfo(c *gin.Context) {
```
Comment (gsanchietti) [inline api/methods/unit.go:146]:
```suggestion
func PushInfo(c *gin.Context) {
```

File: api/main.go (hunk 1 above)
Comment (Tbaile) [reply api/main.go]: Cause we override the info in the files :man_shrugging: 

Changed!

Comment (Tbaile) [reply api/main.go]: Cause we override the info in the files 🤷‍♂️

Changed!

**Review by gsanchietti — APPROVED** [review]

File: api/main.go [hunk 3]
```diff
@@ -140,6 +140,7 @@ func main() {
 
 	// Ingest APIs: receive data from firewalls
 	authorized := router.Group("/ingest", middleware.BasicAuth())
+	authorized.POST("/info", methods.PushInfo)
```
Comment (edospadoni) [inline api/main.go:143]:
```suggestion
	authorized.POST("/info", methods.AddInfo)
```

File: api/methods/unit.go [hunk 4]
```diff
@@ -143,6 +143,36 @@ func GetUnitInfo(c *gin.Context) {
 
 }
 
+func PushInfo(c *gin.Context) {
```
Comment (edospadoni) [inline api/methods/unit.go:146]:
```suggestion
func AddInfo(c *gin.Context) {
```

**Comment by edospadoni** [comment]
for consistency with the other methods, I would change from Push to Add

**Review by edospadoni — APPROVED** [review]

**Pull request merged**